packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1123,22 +1173,14 @@ function simplifyStruct<T>(typeDef: StructType, context: SimplifyContext<T>): Si
 		})
 
 		if (typeof key === 'string') {
-			literalFields.set(key, field)
-		} else if (key.kind === 'literal' && key.value.kind === 'string' && !key.attributes?.length) {
-			literalFields.set(key.value.value, field)
+			fields.push({
+				...field,
+				key: { kind: 'literal', value: { kind: 'string', value: key } },
```

Comment:
How do you get away with not using a map for literal types here? How do you protect against adding a two fields with the same key?

Reply:
Basically, I realized that deduplicating here is mostly obsolete, as I already have a map of runtime keys in line 579. What will happen with multiple matching keys is, it first assigns the first one and later overwrites it with the second one.

The only case I needed to handle is complex keys, which is why I changed the code for those.